MSAL/src/requests/MSALSilentRequest.m

```diff
@@ -227,4 +355,20 @@ - (BOOL)isServerUnavailable:(NSError *)error
     return error.code == MSALErrorUnhandledResponse && responseCode >= 500 && responseCode <= 599;
 }
 
+- (void)updateAppMetadata:(MSIDAppMetadataCacheItem *)appMetadata
+              serverError:(NSError *)serverError
+               cacheError:(NSError **)cacheError
+          completionBlock:(MSALCompletionBlock)completionBlock
+{
+    NSString *subError = serverError.userInfo[MSALOAuthSubErrorKey];
+    if (subError && [subError isEqualToString:MSIDServerErrorClientMismatch])
```

Comment:
nit: please add a comment on what this is doing for future generations :)

Reply:
done